scriptmodules/supplementary/emulationstation/configscripts/mupen64plus.sh

```diff
@@ -0,0 +1,266 @@
+                if echo "$ini_value" | grep -q "("; then
+                    value="${ini_value}${input_id}+)"
+                # if there is already a "+" axis add "-" axis value
+                elif echo "$ini_value" | grep -q ")"; then
+                    value="axis(${input_id}-,${ini_value}"
+                # if there is no ini_value add "+" axis value
+                elif [[ "$input_value" == "1" ]]; then
+                    value="${input_id}+)"
+                else 
+                    value="axis(${input_id}-,"
+                fi
+            elif [[ "$input_value" == "1" ]]; then
+                value="axis(${input_id}+) ${ini_value}"
+            else
+                value="axis(${input_id}-) ${ini_value}"
+            fi
+        elif [[ "$input_type" == "hat" ]]; then
+            if echo "$key" | grep -q "Axis"; then
+                if echo "$ini_value" | grep -q "("; then
+                    value="${ini_value}${input_id} ${dir})"
+                elif echo "$ini_value" | grep -q ")"; then
+                    value="axis(${input_id} ${dir},${ini_value}"
+                elif [ "$dir" == "Up" ] || [ "$dir" == "Left" ] ; then
+                    value="button(${input_id} ${dir},"
+                elif [ "$dir" == "Right" ] || [ "$dir" == "Down" ]; then
+                    value="${input_id} ${dir})"
+                fi
+            fi
+            value="hat(${input_id} ${dir}) ${ini_value}"
+        else
+            # key "X/Y Axis" needs different button naming
+            if echo "$key" | grep -q "Axis"; then
+                # if there is already a "-" value add "+" value
+                if echo "$ini_value" | grep -q "("; then
+                    value="${ini_value}${input_id})"
+                # if there is already a "+" value add "-" value
+                elif echo "$ini_value" | grep -q ")"; then
+                    value="button(${input_id},${ini_value}"
+                elif [[ "$dir" == "Up" ]]; then
+                    value="button(${input_id},"
+                elif [[ "$dir" == "Right" ]]; then
+                    value="${input_id})"
+                elif [[ "$dir" == "Down" ]]; then
+                    value="${input_id})"
+                elif [[ "$dir" == "Left" ]]; then
+                    value="button(${input_id},"
+                fi
+            fi
+            value="button(${input_id}) ${ini_value}"
+        fi
+
+        iniSet "$key" "$value"
+    done
+}
+
+function onend_mupen64plus_joystick() {
+    local device_type=$1
+    local device_name=$2
+    
+    # analog stick sanity check
+    # replace Axis value with DPAD values if there is no Axis 
+    # device setup
+    local bind;
+    if ! grep -q -e "X Axis" -e "Y Axis" /tmp/mp64tempconfig.cfg; then
+        iniGet "DPad U"
+        bind=`echo "$ini_value" | sed -e "s/)/,/g"`
+        iniGet "DPad D"
+        ini_value=`echo "$ini_value" | sed -e "s/button(//g"`
+        ini_value=`echo "$ini_value" | sed -e "s/hat(//g"`
```

Comment:
If you need to removev substring from front / back there are is also specific string manipulation for that.